pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -66,6 +69,21 @@ type topologyPairsMaps struct {
 	podToTopologyPairs map[string]topologyPairSet
 }
 
+// topologyPairsPodSpreadMap combines minMatchMap and topologyPairsMaps to represent
+// (1) how existing pods match incoming pod on its spread constraints
+// (2) minimum match number of each hard spread constraint
+type topologyPairsPodSpreadMap struct {
+	minMatchMap map[string]int32
+	*topologyPairsMaps
```

Comment:
is it necessary to use an existing struct from anti affinity ? I was thinking when we deprecate it, it might be easier to cleanup the code

Reply:
`topologyPairsMaps` is a common data structure, and is not limited to work for affinity only.